include/revng/ABI/ApplyRegisterStateDeductions.h

```diff
@@ -0,0 +1,263 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <array>
+#include <iterator>
+#include <optional>
+#include <span>
+#include <variant>
+
+#include "revng/ADT/STLExtras.h"
+#include "revng/Model/ABI.h"
+#include "revng/Model/Architecture.h"
+#include "revng/Model/Register.h"
+#include "revng/Model/RegisterState.h"
+#include "revng/Support/EnumSwitch.h"
+
+namespace detail {
+
+using NullableRegisterState = std::optional<model::RegisterState::Values>;
```

Comment:
I'd avoid `std::optional` in favor of the already available `Invalid`. I understand `std::optional` is more explicit, but then users might get confused on what to do with `Invalid`.

Reply:
Done (https://github.com/revng/revng/pull/98/commits/3be0a383f482727d8da1bbc82485cc329f2761b6)